app.py

```diff
@@ -378,27 +407,26 @@ def update_graph_parameters(freq, name1, name2, years):
 # <https://community.plot.ly/t/annotation-not-showing-on-dash-dcc-graph/6660>
 
 
-def download_html(freq, name):
-    link_text = f'{freq}_{name}.csv'
-    url = URL(freq, name).datapoints('csv')
+def download_data_html(freq, names, years):
+    link_text = 'csv'
+    url = URL(freq).frame(names, *years)
```

Comment:
NOT TODO: this may need change
```
class URL:

    def __init__(self, freq, name=None):
        self.freq = freq
        self.name = name

    def info(self):
        return (f'{BASE_URL}/api/info'
                f'?freq={self.freq}&name={self.name}')

    def names(self):
        return f'{BASE_URL}/api/names/{self.freq}'

    def datapoints(self, format):
        return (f'{BASE_URL}/api/datapoints'
                f'?freq={self.freq}&name={self.name}&format={format}')

    def frame(self, names, start_date, end_date):
        return (f'{BASE_URL}/api/frame?freq={self.freq}&names={names}'
                f'&start_date={start_date}&end_date={end_date}')

    @property
    def custom_text(self):
        return f'series/{self.name}/{self.freq}'

    @property
    def custom_link(self):
        return f'{BASE_URL}/all/series/{self.name}/{self.freq}'

    @property
    def csv(self):
        return self.datapoints('csv')```